packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1494,7 +1496,7 @@ class TextInput {
           deltas.add(delta);
         }
 
-        _currentConnection!._client.updateEditingValueWithDeltas(deltas);
+        (_currentConnection!._client as DeltaTextInputClient).updateEditingValueWithDeltas(deltas);
```

Comment:
Is there any risk that `_currentConnection!._client` might not be a DeltaTextInputClient?  Is updateEditingStateWithDeltas ever called on any platform when enableDeltaModel is false?  Even if it's not, maybe we need to make it obvious to users in the docs that if they set enableDeltaModel to true, then they must be using an implementation of DeltaTextInputClient and not TextInputClient.

Reply:
On the platform side there is no risk. But yes we can add this to the documentation.

Reply:
We should add an assert here as well! This is one of the problems with casting things around, it adds risk of misuse.

Reply:
I don't think we can accomplish this same thing without breaking the interface API though, so reluctantly, we may have to proceed with something like this even though it is a bit awkward

Reply:
That's right. If a platform implements the delta model, and someone is using a `TextInputClient` with a `TextInputConfiguration` with `enableDeltaModel` set to true then they will be able to reach this case and receive unexpected behavior.

Reply:
Ok, well let's slap tons of warnings and docs and asserts around this to make sure it is at least obvious exactly how to fix it/make it work. Kind of ugly but the cost of being non-breaking.

Reply:
If this were not an opt-in feature, im tempted to just break as most developers using this would be advanced anyways and it should be trivial to copy-paste in a snipped already included in the docs, but I don't think we should break people especially if they don't even need it for their code to keep working.